fix(wallet) send/receive for duplicate transactions

stefandunca opened the pull request:
### Closes [#11960](https://github.com/status-im/status-desktop/issues/11960)
    
It brings consistency when the sender and receiver are in the filter address list. This fixes the case of the sender and receiver in addresses and filters out duplicate entries.

Also

- refactor tests to provide support for owners
- adapt TestGetActivityEntriesWithSameTransactionForSenderAndReceiverInDB
  to the use of owner instead of from

stefandunca committed:
fix(wallet) send/receive for duplicate transactions

Brings consistency in case when sender and receiver are both in the
filter address list. This fixes the case of sender and receiver in
addresses and filters out duplicate entries.

Also

- refactor tests to provide support for owners
- adapt TestGetActivityEntriesWithSameTransactionForSenderAndReceiverInDB
  to the use of owner instead of from

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (28)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 0a0f8252 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/1/) | 2023-09-07 21:10:29 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230907-210604-0a0f82-pr4008.zip) |
| :heavy_check_mark: | 0a0f8252 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/1/) | 2023-09-07 21:11:29 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230907-210609-0a0f82-pr4008.zip) |
| :heavy_check_mark: | 0a0f8252 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/1/) | 2023-09-07 21:13:20 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230907-210604-0a0f82-pr4008.aar) |
| :heavy_check_mark: | 0a0f8252 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/1/) | 2023-09-07 21:37:08 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a6d65c44 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/2/) | 2023-09-08 19:05:55 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230908-190435-a6d65c-pr4008.zip) |
| :heavy_check_mark: | a6d65c44 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/2/) | 2023-09-08 19:07:09 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230908-190435-a6d65c-pr4008.zip) |
| :heavy_check_mark: | a6d65c44 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/2/) | 2023-09-08 19:09:25 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230908-190435-a6d65c-pr4008.aar) |
| :heavy_check_mark: | a6d65c44 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/2/) | 2023-09-08 19:35:46 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fd86faeb | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/3/) | 2023-09-08 21:53:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230908-215110-fd86fa-pr4008.zip) |
| :heavy_check_mark: | fd86faeb | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/3/) | 2023-09-08 21:54:38 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230908-215110-fd86fa-pr4008.zip) |
| :heavy_check_mark: | fd86faeb | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/3/) | 2023-09-08 21:56:24 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230908-215110-fd86fa-pr4008.aar) |
| :heavy_check_mark: | fd86faeb | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/3/) | 2023-09-08 22:22:19 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 22645090 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/4/) | 2023-09-09 19:24:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230909-192304-226450-pr4008.zip) |
| :heavy_check_mark: | 22645090 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/4/) | 2023-09-09 19:24:57 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230909-192304-226450-pr4008.aar) |
| :heavy_check_mark: | 22645090 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/4/) | 2023-09-09 19:25:59 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230909-192304-226450-pr4008.zip) |
| :heavy_multiplication_x: | 22645090 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/4/) | 2023-09-09 19:51:42 | ~28 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bbda7b27 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/5/) | 2023-09-09 19:36:22 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230909-193518-bbda7b-pr4008.zip) |
| :heavy_check_mark: | bbda7b27 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/5/) | 2023-09-09 19:36:47 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230909-193515-bbda7b-pr4008.aar) |
| :heavy_check_mark: | bbda7b27 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/5/) | 2023-09-09 19:37:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230909-193518-bbda7b-pr4008.zip) |
| | | | | | | |
| :heavy_check_mark: | b43adb14 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/6/) | 2023-09-09 19:47:00 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230909-194555-b43adb-pr4008.zip) |
| :heavy_check_mark: | b43adb14 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/6/) | 2023-09-09 19:47:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230909-194555-b43adb-pr4008.aar) |
| :heavy_check_mark: | b43adb14 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/6/) | 2023-09-09 19:48:16 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230909-194555-b43adb-pr4008.zip) |
| :heavy_multiplication_x: | b43adb14 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/5/) | 2023-09-09 20:06:01 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/5//consoleText) |
| :heavy_check_mark: | b43adb14 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/6/) | 2023-09-11 09:04:41 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 148a0993 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/7/) | 2023-09-11 19:55:07 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230911-195405-148a09-pr4008.zip) |
| :heavy_check_mark: | 148a0993 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/7/) | 2023-09-11 19:55:16 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230911-195400-148a09-pr4008.aar) |
| :heavy_check_mark: | 148a0993 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/7/) | 2023-09-11 19:58:29 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230911-195400-148a09-pr4008.zip) |
| :heavy_check_mark: | 148a0993 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/7/) | 2023-09-11 20:23:47 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/7//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | aac76c7b | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/8/) | 2023-09-11 20:26:45 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230911-202450-aac76c-pr4008.aar) |
| :heavy_check_mark: | aac76c7b | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/8/) | 2023-09-11 20:27:30 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230911-202450-aac76c-pr4008.zip) |
| :heavy_check_mark: | aac76c7b | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/8/) | 2023-09-11 20:27:38 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230911-202450-aac76c-pr4008.zip) |
| :heavy_multiplication_x: | aac76c7b | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/8/) | 2023-09-11 20:39:28 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3ced471b | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4008/9/) | 2023-09-11 20:29:16 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230911-202808-3ced47-pr4008.zip) |
| :heavy_check_mark: | 3ced471b | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4008/9/) | 2023-09-11 20:29:22 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230911-202805-3ced47-pr4008.aar) |
| :heavy_check_mark: | 3ced471b | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4008/9/) | 2023-09-11 20:30:39 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230911-202805-3ced47-pr4008.zip) |
| :heavy_check_mark: | 3ced471b | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/9/) | 2023-09-11 21:09:14 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4008/9//consoleText) |

stefandunca committed:
fix(wallet) fix filter by activity type

The activity type filtering was not stable in relation to addresses
filter which was generating unexpected Send/Receive type in the
corner-case when both sender and receiver was in the address list.

Updates status-desktop #11960

Cuteivist reviewed (approved):
Great job.

I tested the branch and works great on transfers where sender == recipient, even for existing data.
